src/db.rs

```diff
@@ -1,67 +1,111 @@
-use crate::dir::Dir;
-use crate::types::{Epoch, Rank};
+use crate::config;
+use crate::dir::{Dir, Epoch, Rank};
+
 use anyhow::{anyhow, bail, Context, Result};
-use std::fs::{self, File};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use std::cmp::Ordering;
+use std::fs::{self, File, OpenOptions};
 use std::io::{self, BufRead, BufReader, BufWriter};
 use std::path::{Path, PathBuf};
 
+pub use i32 as DBVersion;
+
 pub struct DB {
-    path: PathBuf,
-    dirs: Vec<Dir>,
+    data: DBData,
     modified: bool,
+    path: PathBuf,
+    path_old: Option<PathBuf>,
```

Comment:
I don't think we're planning on moving DB locations ever again after this, so might as well make a note to remove this when it's no longer needed (next breaking release).

```suggestion
    // FIXME: remove after next breaking version
    path_old: Option<PathBuf>,
```